src/ai/defaultai.cc

```diff
@@ -5662,18 +5644,20 @@ BuildingNecessity DefaultAI::check_building_necessity(BuildingObserver& bo,
 			}
 		} else if (bo.type == BuildingObserver::Type::kMine) {
 			bo.primary_priority = bo.max_needed_preciousness;
+			// NOCOM the following could be addressed as mines_per_type[bo.mines].total_count as well
+			// NOCOM any reason for leaving this with in_construction + finished
```

Comment:
I see no reason not to use total_count here, so feel free to change it...

Reply:
In the following there are multiple instances similar to this. Shall I go over them as well?

Reply:
You can, I think the total_count() was introduced later, this is why it is this way

Reply:
Ok. Thanks for the explanation.